examples/pytorch/rrn/train_sudoku.py

```diff
@@ -0,0 +1,143 @@
+from sudoku_data import sudoku_dataloader
+import argparse
+from sudoku import SudokuNN
+import torch
+from torch.optim import Adam
+import os
+import numpy as np
+import logging
+
+
+def accuracy(preds, target):
+    batch_size = preds.size(0)
+    right_count = 0
+    for i in range(batch_size):
+        if torch.all(torch.eq(preds[i], target[i])).__bool__():
+            right_count += 1
+    return right_count / batch_size
+
+
+def main(args):
+    device = torch.device('cuda' if torch.cuda.is_available() else "cpu")
```

Comment:
We should let user decide whether to use cpu/gpu with given id.
Please refer to https://github.com/dmlc/dgl/blob/master/examples/pytorch/gcn/train.py.